test/functional/page_objects/discover_page.ts

```diff
@@ -51,19 +49,20 @@ export function DiscoverPageProvider({ getService, getPageObjects }: FtrProvider
     }
 
     public async saveSearch(searchName: string) {
-      log.debug('saveSearch');
       await this.clickSaveSearchButton();
-      await testSubjects.setValue('savedObjectTitle', searchName);
-      await testSubjects.click('confirmSaveSavedObjectButton');
-      await header.waitUntilLoadingHasFinished();
+      // preventing an occasional flakiness when the saved object wasn't set and the form can't be submitted
+      await retry.waitFor(`saved search can be persisted name ${searchName}`, async () => {
+        await testSubjects.setValue('savedObjectTitle', searchName);
+        await testSubjects.click('confirmSaveSavedObjectButton');
+        return (await header.isGlobalLoadingIndicatorVisible()) === true;
```

Comment:
I thought we might need to wait for popup to disappear
`return (await testSubjects.exists('confirmSaveSavedObjectButton', { timeout: 5000 })) === false`

Is there a chance that popup will remain in place and `header.isGlobalLoadingIndicatorVisible()` will return `true` ?

Reply:
good point  👍 , I will take a closer look here!

Reply:
I'm wondering if we can always catch the loading indicator after clicking save.  Could it be that in some cases the save is quick enough that we never see the loading indicator?

Also, if we do find the loading indicator, I think we should then wait for it to be hidden before returning from saveSearch method.

Reply:
I've changed the code to check for the save button not to be disabled, this should catch the occasional
flakiness when for whatever reason the title wasn't entered, and the save button stayed disabled therefore

Reply:
flaky test runner says, it's ok this way: https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/712/